bump compose version to v2.23.3 and add release notes (#18769)
docker/docs: content/compose/release-notes.md

For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).

## 2.23.3
{{< release-date date="2023-11-22" >}}

### Update
- Dependencies upgrade: bump buildx to v0.12.0

## 2.23.2
{{< release-date date="2023-11-21" >}}

### Update
- Dependencies upgrade: bump buildkit 0.12.3 
- Dependencies upgrade: bump docker 24.0.7 
- Dependencies upgrade: bump cli 24.0.7 
- Dependencies upgrade: bump 1.20.2

### Bug fixes and enhancements

> - Support `builds.tags` with `push` command

Reviewer: ```suggestion
- Compose now supports `builds.tags` with `push` command.
```

> - Compose watch now re-build service images at startup

Reviewer: ```suggestion
- Compose Watch now re-builds service images at startup.
```

> - Fixed issue, now `--remove-orphans` don't manage disabled services as orphaned

Reviewer: ```suggestion
- Now `--remove-orphans` doesn't manage disabled services as orphaned.
```

> - Display `Building` output log only if there is at least a service to build

Reviewer: ```suggestion
- Compose displays `Building` output log only if there is at least one service to build.
```

## 2.23.1
{{< release-date date="2023-11-16" >}}

### Update
- Dependencies upgrade: bump compose-go to v1.20.1

### Bug fixes and enhancements
- Align with OCI artifact best practices
- Introduce --resolve-image-digests for publish to seal service images
- Improve watch configuration logging

> - Reject compose file using secrets|configs.driver or template_driver

Reviewer: ```suggestion
- Compose now rejects a Compose file using `secrets|configs.driver` or `template_driver`.
```

> - Fail start if depependency is missing by @ndeloof in #11110

Reviewer: ```suggestion
- Compose now fails to start if a dependency is missing. 
```

> - Fixed SIGTERM support to stop/kill stack

Reviewer: ```suggestion
- Fixed SIGTERM support to stop/kill stack.
```

> - Fixed a --hash regression

Reviewer: ```suggestion
- Fixed a `--hash` regression.
```

> - Fixed "Application failed to start after update" when an external network is on a watched service

Reviewer: ```suggestion
- Fixed "Application failed to start after update" when an external network is on a watched service.
```

> - Fixed `--pull` documentation

Reviewer: ```suggestion
- Fixed `--pull` documentation.
```

> - Fixed display by adding newline in cmd/compose/build.go

Reviewer: ```suggestion
- Fixed display by adding newline in cmd/compose/build.go.
```

> - Render quiet after filtering applied

Reviewer: ```suggestion
- Compose is rendered quiet after filtering applied.
```

> - Strip project prefix from docker-compose up output

Reviewer: ```suggestion
- Stripped project prefix from docker-compose up output.
```
